### PR TITLE
HCAP-1022 | Merge local and context reducers

### PR DESCRIPTION
**Pull request by arranfw**
https://freshworks.atlassian.net/browse/HCAP-1022

### CODE REVIEW

File: client/src/components/generic/DebounceTextField.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const DebounceTextField = ({ time, onDebounce, onChange, ...props }) => {
 
   const handleOnChange = (element) => {
     setText(element.target.value);
-    onChange(element);
+    onChange && onChange(element);
```
Comment (dbayly-freshworks) [inline client/src/components/generic/DebounceTextField.js:20]: Are there issues with onChange not being defined?

File: client/src/pages/private/ParticipantTable.js [hunk 2]
```diff
@@ -282,22 +257,21 @@ const ParticipantTable = () => {
...
-        reducerState.emailFilter || '',
-        reducerState.order.field,
-        reducerState.order.direction,
-        reducerState.siteSelector,
+      const { data, pagination: newPagination } = await fetchParticipants(
+        pagination.page * pageSize,
+        filter.locationFilter?.value || '',
+        filter.fsaFilter?.value || '',
+        filter.lastNameFilter?.value || '',
+        filter.emailFilter?.value || '',
```
Comment (dbayly-freshworks) [inline client/src/pages/private/ParticipantTable.js:265]: Threre's still a fair bit of duplication here. I wonder if creating a prop type could be useful here.

File: client/src/components/generic/DebounceTextField.js (hunk 1 above)
Comment (fwpushan) [reply client/src/components/generic/DebounceTextField.js:20]: Yes, I've same question

**Review by fwpushan — DISMISSED** [review]
LGTM

File: client/src/components/generic/DebounceTextField.js (hunk 1 above)
Comment (arranfw) [reply client/src/components/generic/DebounceTextField.js:20]: This component handle it's own textfield state so the onchange/value props aren't really needed. I removed those values from the implementations of this component and only use `defaultValue`/`onDebounce` props. Removing `onChange` cause an error here

File: client/src/pages/private/ParticipantTable.js (hunk 2 above)
Comment (arranfw) [reply client/src/pages/private/ParticipantTable.js:265]: Totally, this gets alleviated a bit in https://github.com/bcgov/hcap/pull/617

**Review by dbayly-freshworks — APPROVED** [review]
Perfectly balanced as all PRs should be.

**Pull request merged**